### PR TITLE
debezium/dbz#1926 Resolve component name based on the class FQN.

### PR DESCRIPTION
**Pull request by mfvitale**
Fixes debezium/dbz#1926

## Description
This pull request introduces a new, extensible mechanism for generating human-readable display names for Debezium components based on their class names and types. The main improvement is the addition of the `DisplayNameResolver` utility, which replaces a hardcoded switch statement with an algorithmic approach, and the integration of this resolver throughout the codebase. Several connector metadata providers are also updated to explicitly provide display names for their main components. Comprehensive unit tests are added to ensure the correctness of the new resolver.

**Display Name Generation Improvements:**

* Added the new `DisplayNameResolver` class, which algorithmically derives display names from class names and component types, handling brand names, CamelCase splitting, and type suffixes. This replaces the previous hardcoded mapping.
* Removed the old `getDisplayNameForConnectorClass` method and updated `ComponentDescriptor` to use the new resolver when a display name is not provided. [[1]](diffhunk://#diff-796f7e8cc531e69ff3f9f298322c7d56a58a42b8b03bd7a71fb2f15442bb1a65L49-R60) [[2]](diffhunk://#diff-796f7e8cc531e69ff3f9f298322c7d56a58a42b8b03bd7a71fb2f15442bb1a65L148-L159)

**API and Usage Updates:**

* Updated the `ComponentMetadataFactory` to allow passing an explicit display name, and refactored its usage in connector metadata providers.
* Updated `MariaDbMetadataProvider`, `MongoDbMetadataProvider`, and `MySqlMetadataProvider` to provide explicit display names for their main connector components. [[1]](diffhunk://#diff-1cebbbbc2c503da7ba881ad60b761e47f492746d2c7015629d435813c0cf0890L28-R28) [[2]](diffhunk://#diff-015d7173fc8ff31f5edac51e8d3c3dab95c9b51a4fe2f0d04cb2ef6906b8550bL29-R29) [[3]](diffhunk://#diff-01dbad5c6497b0690807648d5fb3b4fd7ec5264197a3ede57c9d7d0eed3fca13L29-R29)

**Testing Enhancements:**

* Added a comprehensive test suite for `DisplayNameResolver`, covering various connector types, brand name handling, CamelCase splitting, and edge cases.

## PR Checklist
- [X] I have read the [contribution guidelines](https://github.com/debezium/debezium/blob/main/CONTRIBUTING.md) and the [governance document](https://github.com/debezium/governance/blob/main/GOVERNANCE.md) on PR expectations.
- [X] Minimal changes to code not directly related to your change (e.g. no unnecessary formatting changes or refactoring to existing code)
- [X] One feature/change per PR unless tightly coupled
- [X] Do a rebase on upstream `main`

### CODE REVIEW

File: debezium-config/src/main/java/io/debezium/metadata/DisplayNameResolver.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+            SERVER_SINK_TYPE, List.of("ChangeConsumer", "SinkConsumer"),
+            CONVERTER_TYPE, List.of("Converter"),
+            CUSTOM_CONVERTER_TYPE, List.of("Converter"));
+
+    private static final Map<String, String> TYPE_DISPLAY_SUFFIXES = Map.of(
+            SOURCE_CONNECTOR_TYPE, "Connector",
+            SINK_CONNECTOR_TYPE, "Sink Connector",
+            SERVER_SINK_TYPE, "Server Sink",
+            CONVERTER_TYPE, "Converter",
+            CUSTOM_CONVERTER_TYPE, "Custom Converter");
```
Comment (Naros) [inline debezium-config/src/main/java/io/debezium/metadata/DisplayNameResolver.java:67]: What about Transformations and/or Post Processors? Should those be included?

Comment (mfvitale) [reply debezium-config/src/main/java/io/debezium/metadata/DisplayNameResolver.java:67]: For them we just use the name without any other prefix/suffix. Honestly I was in doubt also for `Server Sink` to just keep the name, but then I leaved it. I do not have a strong opinion. We can still change them after if we see that it not looks good on the UI. WDYT?

Comment (mfvitale) [reply debezium-config/src/main/java/io/debezium/metadata/DisplayNameResolver.java:67]: @Naros Any other input on this?

Comment (Naros) [reply debezium-config/src/main/java/io/debezium/metadata/DisplayNameResolver.java:67]: I completely agree. I think we can iterate on it after we see it in practice. The only point I would highlight is that for sinks, in particular those that use the Kafka sink bridge for DS, we'll need to think about how that is going to work in practice when we get there.

**Review by Naros — APPROVED** [review]

**Comment by Naros** [comment]
LGTM @mfvitale, feel free to merge yourself if there is no other input you need.

File: debezium-config/src/main/java/io/debezium/metadata/DisplayNameResolver.java (hunk 1 above)
Comment (mfvitale) [reply debezium-config/src/main/java/io/debezium/metadata/DisplayNameResolver.java:67]: > in particular those that use the Kafka sink bridge for DS, we'll need to think about how that is going to work in practice when we get there.

You mean for the configuration that are passthrought?
